src/fish_tests.cpp

```diff
@@ -6828,23 +6812,6 @@ static void test_re_basic() {
     }
     do_test(join_strings(matches, L',') == L"AA,CC,11");
     do_test(join_strings(captures, L',') == L"A,C,1");
```

Comment:
this should be in the commit that ports make_anchored. Will fix.
Actually, the make_anchored commit port doesn't compile because C++ is still using it.
Totally not worth adding a temporary FFI wrapper, so let's do a merge commit instead